nova/scheduler/weights/resize_same_host.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) 2020 OpenStack Foundation
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+"""
+RAM Weigher.  Weigh hosts by their RAM usage.
+
+The default is to spread instances across all hosts evenly.  If you prefer
+stacking, you can set the 'ram_weight_multiplier' option to a negative
+number and the weighing has the opposite effect of the default.
+"""
+import nova.conf
+from nova.exception import ObjectActionError
+from nova.objects.flavor import Flavor
+from nova.scheduler import weights
+
+CONF = nova.conf.CONF
+
+
+class PreferSameHostOnResizeWeigher(weights.BaseHostWeigher):
+    minval = 0
+    # `id` might change if flavors are edited, but does not imply a change of
+    # flavor.
+    # a change of `projects` a flavor is configured for is not indicative of
+    # any parameter changes.
+    # `description` is strictly user-facing metadata.
+    ignored_flavor_fields = ['id', 'projects', 'description']
+
+    def weight_multiplier(self):
+        """Override the weight multiplier."""
+        return CONF.filter_scheduler.prefer_same_host_resize_weight_multiplier
+
+    def _is_resize(self, cur_flavor, dest_flavor):
+        """Return True if any parameters change between the two flavors"""
+        for field in Flavor.fields:
+            if field in self.ignored_flavor_fields:
+                continue
+            if hasattr(cur_flavor, field) != hasattr(dest_flavor, field):
+                return True
+            try:
+                if getattr(cur_flavor, field) != getattr(dest_flavor, field):
+                    return True
+            except AttributeError:
+                continue
+            except ObjectActionError as e:
+                if 'obj_load_attr' in str(e):
+                    continue
+                raise e
+        return False
```

Comment:
Why can't we compare the `id` field of both flavors instead? If we updated the flavor and the `name`/`flavorid` field are still the same, the `id` field should still be different. In that case, we must assume there's a change and act as if it's a resize.
While this code might work, too, I think we should keep it simple.

Reply:
I considered id fields to be insufficient because they may change. I can change it, and the consequence would be that any id change would be a resize and thus even if only the description changed, the host would be preferably scheduled to the same host... Sounds insignificant. Will do.